universe/base.go

```diff
@@ -245,6 +257,60 @@ func (a *MintingArchive) RegisterIssuance(ctx context.Context, id Identifier,
 	return issuanceProof, nil
 }
 
+func (a *MintingArchive) getPrevAssetSnapshot(ctx context.Context,
+	uniID Identifier, newProof proof.Proof) (*proof.AssetSnapshot, error) {
+
+	// If this is a genesis proof, then there is no previous asset (and
+	// therefore no previous asset snapshot).
+	if newProof.Asset.HasGenesisWitness() {
+		return nil, nil
+	}
+
+	// Query for proof associated with the previous asset.
+	prevID := newProof.Asset.PrimaryPrevID()
```

Comment:
Since this _can_ be nil in case there is no witness (shouldn't happen, but better be safe than sorry), we should check and return an error (invalid proof), otherwise this can trigger a panic if I send an invalid proof to the universe.

Reply:
Or maybe we shouldn't just return `nil` but have it return an error instead. Better be ultra cautious in any validation code.